src/game/logic/object/weapon.cpp

```diff
@@ -954,6 +1520,214 @@ void WeaponTemplate::Deal_Damage_Internal(ObjectID source_id,
+                if (it->frame >= frame2) {
+                    if (Is_2D_Dist_Squared_Less_Than(pos, &it->location, radius_sqr)) {
+                        count++;
+                    }
+                }
+            }
 
-#endif
+            if (count < m_historicBonusCount - 1) {
+                HistoricWeaponDamageInfo info(frame, *pos);
+                m_historicDamage.push_back(info);
+            } else {
+                g_theWeaponStore->Create_And_Fire_Temp_Weapon(m_historicBonusWeapon, source, pos);
+                m_historicDamage.clear();
+            }
+        }
+
+        Object *victim;
+
+        if (victim_id != OBJECT_UNK) {
+            victim = g_theGameLogic->Find_Object_By_ID(victim_id);
+        } else {
+            victim = nullptr;
+        }
+
+        if (victim != nullptr) {
+            pos = victim->Get_Position();
+        }
+
+        DamageType damage = Get_Damage_Type();
+        DeathType death = Get_Death_Type();
+        ObjectStatusTypes status = Get_Damage_Status_Type();
+
+        if (Get_Projectile_Template() == nullptr || is_projectile_detonation) {
+            float primary_radius = Get_Primary_Damage_Radius(bonus);
+            float secondary_radius = Get_Secondary_Damage_Radius(bonus);
+            float primary_damage = Get_Primary_Damage(bonus);
+            float secondary_damage = Get_Secondary_Damage(bonus);
+            int mask = Get_Affects_Mask();
+
+            captainslog_dbgassert(secondary_radius > primary_radius || secondary_radius == 0.0f,
+                "secondary radius should be >= primary radius (or zero)");
+
+            float primary_radius_sqr = GameMath::Square(primary_radius);
+            float max = std::max(primary_radius, secondary_radius);
+            SimpleObjectIterator *iter;
+            float numeric;
+            Object *obj;
+
+            if (max <= 0.0f) {
+                iter = nullptr;
+                numeric = 0.0f;
+                obj = victim;
+                if ((mask & AFFECTS_SUICIDE) != 0) {
+                    DamageInfo info;
+                    info.m_in.m_damageType = damage;
+                    info.m_in.m_deathType = death;
+                    info.m_in.m_sourceID = source_id;
+                    info.m_in.m_playerMask = 0;
+                    info.m_in.m_damageStatusType = status;
+                    info.m_in.m_amount = 999999.0f;
+                    source->Attempt_Damage(&info);
+                }
+            } else {
+                iter =
+                    g_thePartitionManager->Iterate_Objects_In_Range(pos, max, FROM_BOUNDINGSPHERE_3D, nullptr, ITER_FASTEST);
+                obj = iter->First_With_Numeric(&numeric);
+            }
+
+            MemoryPoolObjectHolder holder(iter);
+
+            for (;;) {
+                if (obj == nullptr) {
+                    return;
+                }
+
+                bool suicide = false;
+                DamageInfo info;
+                float angle;
```

Comment:
Declaration can be moved down to `angle = Get_Radius_Damage_Angle();`

Reply:
Can't be done because of the gotos above.